recipes/xerces-c/all/conanfile.py

```diff
@@ -12,15 +13,28 @@ class XercesCConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "http://xerces.apache.org/xerces-c/index.html"
     license = "Apache-2.0"
-    exports_sources = ["CMakeLists.txt"]
-    generators = "cmake"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake", "cmake_find_package_multi"
```

Comment:
```suggestion
    generators = "cmake", "cmake_find_package"
```

Reply:
Changing `cmake_find_package_multi` to `cmake_find_package` results in a build error:
```log
-- Build files have been written to: /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/44e11b284c776043657ad061b810d0f3d9f52014/build_subfolder
[  1%] Generating xercesc_messages.c
[  1%] Generating root.res
[  1%] Generating root_res.c
Error renaming from "xercesc_messages_3_2_dat.c" to "xercesc_messages.c": No such file or directory
make[2]: *** [source_subfolder/src/CMakeFiles/xerces-c.dir/build.make:74: source_subfolder/src/xercesc_messages.c] Error 1
make[1]: *** [CMakeFiles/Makefile2:141: source_subfolder/src/CMakeFiles/xerces-c.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
xerces-c/3.2.3: 
xerces-c/3.2.3: ERROR: Package '44e11b284c776043657ad061b810d0f3d9f52014' build failed
xerces-c/3.2.3: WARN: Build folder /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/44e11b284c776043657ad061b810d0f3d9f52014
ERROR: xerces-c/3.2.3: Error in build() method, line 126
        cmake.build()
        ConanException: Error 2 while executing cmake --build '/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/44e11b284c776043657ad061b810d0f3d9f52014/build_subfolder' '--' '-j16'
```
Have to investigate why that is.

Reply:
let's use `cmake_find_package_multi` going forward, see also #7153

Reply:
Ok, cool. I'm fine with that.

Reply:
It doesn't make sense to use cmake_find_package_multi here.

Reply:
Ok, I think I get the whole thing about when which FindPackage.cmake file is used or not.

I thought it wouldn't make a big difference whether the FindICU from xerces-c or from conan is used because I assumed they would be the same. I didn't expect xerces-c to define custom variables in their FindICU file.

So the problem I'm seeing with the failing build is due to xerces-c using `ICU_GENCMN_EXECUTABLE`, for example, to refer to ICU's `gencmn` executable.  
Do you have any smart ideas on how this could be solved? Because off the top of my hat, I could only come up with defining these variables (i.e. `ICU_GENCMN_EXECUTABLE` and so on) manually in the conanfile.py like so:
```py
def _configure_cmake(self):
    ...
    if "icu" in (self.options.transcoder, self.message_loader):
        self._cmake.definitions["ICU_GENCMN_EXECUTABLE"] = "gencmn" 
        ...
```
But I'm not quite sure if this would pick up the `gencmn` executable from Conan's ICU or from my system.   
Maybe there is a better way to get executable CMake targets from conan?

<br>

> Note that cross-build is broken anyway in this case, because icu is not in build requirements.

Does that mean we need to add ICU to build_requires as well? Frankly, I haven't thought about cross-builds at all and I don't actually know all of the things that need to be considered for cross-builds.

Reply:
https://github.com/FMeinicke/conan-center-index/pull/2 fixes cross-build and allow to use Find CMake files generated by conan instead of custom FindICU.cmake of upstream or FindCURL.cmake from CMake installation.

Reply:
Thank you very much for your help. I tested the PR on my system as well and it works like a charm!

I've got just one question regarding cross-building: Why is it that it's not enough to have ICU in the requirements? Why does it have to be added to the build requirements as well? Does this have to do with the gencmn executables that are used when `message_loader=icu`?

Reply:
> Does this have to do with the gencmn executables that are used when message_loader=icu?

Yes, when cross-building, icu executables are executed on build machine.

Reply:
Ah, right! The ICU executables have to be present on the build machine as well. That's what I forgot.

Thanks for clearing this up!